### PR TITLE
Add non relative import and build using esbuild

### PR DESCRIPTION
**Pull request by beam41**
# Description

I'm reviving #63 but I use esbuild instead because tsx built on top of that. (It also fast)

- have to created build script because I need plugin that mark packages as external
- `cross-env` for someone that test production build on windows (like me, lol)
- change import from `.js` => `.ts`, it's less confusing, esbuild doesn't care anyway.
- also resolve non relative import on test file.
- @3raphat I added you to co-author 😉

## Type of change

- [x] Refactor or other

# Checklist:

- [x] I have run `pnpm format` and my code don't have any linting issues
  
  
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: test/tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,11 @@
   "extends": "@tsconfig/node18",
   "compilerOptions": {
     "noEmit": true,
-    "moduleResolution": "nodenext"
+    "moduleResolution": "nodenext",
```
Comment (Leomotors) [inline test/tsconfig.json:5]: We can set this to `bundler` and no longer need to add `.ts` to imports.

Comment (Leomotors) [reply test/tsconfig.json]: Added in TS 5.0

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-0.html#--moduleresolution-bundler

**Review by dtinth — APPROVED** [review]
looks great!

i’m fix the conflict and tests to get this merged asap, because other parts of the codebase can benefit from this greatly.

**Comment by beam41** [comment]
Thanks for fix kub. @dtinth

**Comment by dtinth** [comment]
@beam41 as you are one of the top contributors, i sent you an invitation to give write access to this repo krub. please check your email

**Pull request merged**